src/Agent.Worker/TestResults/ResultsCommandExtension.cs

```diff
@@ -518,6 +195,111 @@ private void AddTargetBranchInfoToRunCreateModel(RunCreateModel runCreateModel,
             }
         }
 
+        private TestRunContext CreateTestRunContext()
+        {
+            string releaseUri = null;
+            string releaseEnvironmentUri = null;
+
+            string teamProject = _executionContext.Variables.System_TeamProject;
+            string owner = _executionContext.Variables.Build_RequestedFor;
+            string buildUri = _executionContext.Variables.Build_BuildUri;
+            int buildId = _executionContext.Variables.Build_BuildId ?? 0;
+            string pullRequestTargetBranchName = _executionContext.Variables.System_PullRequest_TargetBranch;
+            string stageName = _executionContext.Variables.System_StageName;
+            string phaseName = _executionContext.Variables.System_PhaseName;
+            string jobName = _executionContext.Variables.System_JobName;
+            int stageAttempt = _executionContext.Variables.System_StageAttempt ?? 0;
+            int phaseAttempt = _executionContext.Variables.System_PhaseAttempt ?? 0;
+            int jobAttempt = _executionContext.Variables.System_JobAttempt ?? 0;
+
+            //Temporary fix to support publish in RM scenarios where there might not be a valid Build ID associated.
+            //TODO: Make a cleaner fix after TCM User Story 401703 is completed.
+            if (buildId == 0)
+            {
+                _platform = _configuration = null;
+            }
+
+            if (!string.IsNullOrWhiteSpace(_executionContext.Variables.Release_ReleaseUri))
+            {
+                releaseUri = _executionContext.Variables.Release_ReleaseUri;
+                releaseEnvironmentUri = _executionContext.Variables.Release_ReleaseEnvironmentUri;
+            }
+
+            // If runName is not provided by the task, then create runName from testRunner name and buildId.
+            string runName = String.IsNullOrWhiteSpace(_runTitle)
+                ? String.Format("{0}_TestResults_{1}", _testRunner, buildId)
+                : _runTitle;
+
+            StageReference stageReference = new StageReference() { StageName = stageName, Attempt = Convert.ToInt32(stageAttempt) };
+            PhaseReference phaseReference = new PhaseReference() { PhaseName = phaseName, Attempt = Convert.ToInt32(phaseAttempt) };
+            JobReference jobReference = new JobReference() { JobName = jobName, Attempt = Convert.ToInt32(jobAttempt) };
+            PipelineReference pipelineReference = new PipelineReference()
+            {
+                PipelineId = buildId,
+                StageReference = stageReference,
+                PhaseReference = phaseReference,
+                JobReference = jobReference
+            };
```

Comment:
This TestRunContext works in the old path, as previous one didn't have any StageReference or PipelineReference.?

Reply:
PipelineReference was not present in TestRunContext
But it was present in TestRunData and we used to create the same using runContext information
so the behaviour remains the same

old- https://github.com/microsoft/azure-pipelines-agent/blob/master/src/Agent.Worker/TestResults/ResultsCommandExtension.cs#L257

new- https://github.com/tanu-shree136/azure-pipelines-agent/blob/users/tanu/ptrLib/src/Agent.Worker/TestResults/Legacy/LegacyTestRunDataPublisher.cs#L185